interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/Validations.scala

```diff
@@ -89,4 +91,30 @@ object Validations {
       .map(_ => parameter)
   }
 
+  def validateDictEditorParameters(
+      parameters: List[NodeParameter],
+      paramDefMap: Map[String, Parameter],
+      dictRegistry: DictRegistry
+  )(
+      implicit nodeId: NodeId
+  ): ValidatedNel[PartSubGraphCompilationError, Unit] = {
+    parameters
+      .flatMap { param =>
+        paramDefMap.get(param.name).map(definition => validateDictParameter(definition, param, dictRegistry))
+      }
+      .sequence
+      .map(_ => ())
+  }
+
+  private def validateDictParameter(definition: Parameter, parameter: NodeParameter, dictRegistry: DictRegistry)(
+      implicit nodeId: NodeId
+  ): ValidatedNel[PartSubGraphCompilationError, Unit] = definition.editor match {
+    case Some(DictParameterEditor(dictId)) =>
+      dictRegistry
+        .labelByKey(dictId, parameter.expression.expression)
```

Comment:
How it would work on the engine side? We don't want to engine query the registry - take a look at `DictRegistry.toEngineRegistry`

Reply:
`keyByLabel` is the one that cannot be used on engine side, so it should be fine, unless i'm misunderstanding something